### PR TITLE
fix(redis): import redis pipeline using full path

### PR DESCRIPTION
**Pull request by olksdr**
Currently  Sentry uses [`redis-py-cluster==2.1.0`](https://github.com/getsentry/sentry/blob/fd84edc46ad29083c7fe69ff48a159f1ae8ea85a/requirements-frozen.txt#L91) which does not expose the `ClusterPipeline` as a top level object. The fix was introduced in version `2.1.1` and another fix in `2.1.2` see https://github.com/Grokzen/redis-py-cluster/blob/master/docs/release-notes.rst

And getting following error 
```python
09:05:04 [INFO] sentry.plugins.github: apps-not-configured
Traceback (most recent call last):
  File "/Users/olksdr/dev/sentry/.venv/lib/python3.8/site-packages/sentry_sdk/integrations/redis.py", line 123, in setup_once
    _patch_rediscluster()
  File "/Users/olksdr/dev/sentry/.venv/lib/python3.8/site-packages/sentry_sdk/integrations/redis.py", line 88, in _patch_rediscluster
    pipeline_cls = rediscluster.ClusterPipeline
AttributeError: module 'rediscluster' has no attribute 'ClusterPipeline'
```

This PR makes sure to use the full path to import `rediscluster.pipeline.ClusterPipeline` which always available in any version now. 

See errors in integration tests on https://github.com/getsentry/relay/pull/1413 and https://github.com/getsentry/relay/pull/1414 where the sentry `master` branch is used to run those tests with newly updated python sentry-sdk dependency. 

Related changes which "broke" the sdk for redis-py-cluster==2.1.0 were introduced in https://github.com/getsentry/sentry-python/pull/1543

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
@olksdr Thanks for this!

Let's wait for SDK owners' approval before merging.

**Comment by sl0thentr0py** [comment]
thx @olksdr I'm adding a test matrix entry from #1562 that captures this.

**Review by sl0thentr0py — APPROVED** [review]

**Pull request merged**
